core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -693,6 +717,38 @@ private static void removeSharePartitionFromCache(
         }
     }
 
+    /**
+     * The handler to update the failed share fetch request metrics.
+     *
+     * @return A BiConsumer that updates the failed share fetch request metrics.
+     */
+    private BiConsumer<Collection<TopicIdPartition>, Boolean> failedShareFetchMetricsHandler() {
+        return (topicIdPartitions, allTopicPartitionsFailed) -> {
+            // Update failed share fetch request metric.
+            topicIdPartitions.forEach(topicIdPartition ->
+                brokerTopicStats.topicStats(topicIdPartition.topicPartition().topic()).failedShareFetchRequestRate().mark());
```

Comment:
This seems like another instance where `topicIdPartition.topicPartition().topic()` can just be `topicIdPartition.topic()`.

Reply:
Done.